src/memory.c

```diff
@@ -129,6 +129,20 @@ void *mem_realloc(void *old_ptr, size_t size)
 	return res;
 }
 
+char *xstrdup (const char* str)
+{
+  char *res = strdup(str);
+
+  if (res == NULL)
+  {
+    fprintf (stderr, "strdup: %s\n", strerror (ENOMEM));
+
+    exit (-1);
+  }
```

Comment:
OK so now I think we only have some coding style nitpicks left to deal with.

First, the bracket att line 136 should be in the end of line 135 (with a space before it of course).
Second, there should be no space between `fprintf` and the left paren after it.

And finally, please squash your fixes so there's only a single commit for "Added xstrdup() wrapper and replaced accordingly" and force-push it here.

Reply:
I think @Shashank-Rangu was following the coding style of `pp.c`, which isn't our project's. So `xstrdup` should be rewritten using the style of a nearby function in `memory.c` now - this includes not only indentation, but also e.g. use of `error();` in place of `exit (-1);`.

> please squash your fixes so there's only a single commit for "Added xstrdup() wrapper and replaced accordingly" and force-push it here.

Right. There should be 3 commits here (like this PR originally had), not 4.

Reply:
> OK so now I think we only have some coding style nitpicks left to deal with.
> 
> First, the bracket att line 137 should be in the end of line 136 (with a space before it of course). Second, there should be no space between `fprintf` and the left paren after it.
> 
> And finally, please squash your fixes so there's only a single commit for "Added xstrdup() wrapper and replaced accordingly" and force-push it here.

Done. Please check if everything is up to the mark.